misc/python/materialize/output_consistency/generators/expression_generator.py

```diff
@@ -138,25 +160,77 @@ def _generate_args_for_operation(
 
         args = []
 
-        for arg_index in range(0, number_of_args):
+        must_use_aggregation = False
+        for arg_index in range(FIRST_ARG_INDEX, number_of_args):
             param = operation.params[arg_index]
-            arg = self.generate_arg_for_operation_param(param, storage_layout)
+            # nesting_level was already incremented before invoking this function
+            arg = self._generate_arg_for_param(
+                operation,
+                param,
+                arg_index,
+                storage_layout,
+                must_use_aggregation,
+                nesting_level,
+            )
             args.append(arg)
 
+            if arg.is_aggregate:
+                # first argument is aggregated, therefore all must be aggregated (or constant, currently not supported)
+                must_use_aggregation = True
+
         if (
             self.config.avoid_expressions_expecting_db_error
             and try_number <= 50
             and operation.is_expected_to_cause_db_error(args)
         ):
+            # retry
             return self._generate_args_for_operation(
-                operation, storage_layout, try_number=try_number + 1
+                operation,
+                storage_layout,
+                nesting_level=nesting_level,
+                try_number=try_number + 1,
             )
 
         return args
 
-    def generate_arg_for_operation_param(
-        self, param: OperationParam, storage_layout: ValueStorageLayout
+    def _generate_arg_for_param(
+        self,
+        operation: DbOperationOrFunction,
+        param: OperationParam,
+        arg_index: int,
+        storage_layout: ValueStorageLayout,
+        must_use_aggregation: bool,
+        nesting_level: int,
     ) -> Expression:
+        if self.config.generate_nested_expressions and (
+            must_use_aggregation or self.randomized_picker.random_boolean(0.2)
+        ):
+            if must_use_aggregation:
+                allow_aggregation_in_arg = True
+            else:
+                # currently allow an aggregation function as argument if all applies:
+                # * the operation is not an aggregation (nested aggregations are impossible)
+                # * it is first param (all consecutive params with require aggregation)
```

Comment:
Technically speaking, you can stuff all the other non-aggregate parameters in a list and then emit it as a `GROUP BY` clause at the end  of the query. But this may be implemented later if there are any QA-worthy developments in aggregation specifically.